### PR TITLE
[V1] Cache logits buffer in sampler

### PR DESCRIPTION
**Pull request by WoosukKwon**

### CODE REVIEW

**Comment by ywang96** [comment]
As we discussed offline - this alone doesn't fix the random sampling issue.
